nucypher/characters/base.py

```diff
@@ -33,7 +34,7 @@ class Character(Learner):
 
     def __init__(
         self,
-        domain: str,
+        domain: Union[str, TACoDomain],
```

Comment:
Any reason we don't just take a `TACoDomain` instead of a Union that includes str?

Reply:
That will work, doing so will reduce the usage of domains as strings even more, perhaps only exposed by config.   The initial thought was backwards-compatibility but perhaps that's misplaced here.

Reply:
Just wondering how "integrated" the object can be...if you look at the code and it doesn't make sense in some areas, that's fair.

Reply:
✔️